### ISSUE TITLE
Exception if changing widget-specific properties when no selection

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#737]
@billreed63, @peller, @childsb, @jhpedemonte 

If nothing is selected and you change any of the widget-specific properties (left over from previous selection), you get an exception in getParent() within widget.js:

```
app/dojo/../davinci/ve/widget.js:1512 Uncaught TypeError: Cannot read property 'parentNode' of null
```

To reproduce:
- New HTML file
- Add a dijit.form.Button
- Bring up Widget-Specific properties
- Close the file
- New HTML file
- NOTE: Widget-Specific properties will reshow the (now inaccurate) properties from Button even though there is no Button on the page
- Change any of the widget-specific properties

Result=> exception as listed above

There are at least a couple of things that need to be addressed here. First, Properties palette shouldn't show properties for a widget that is not currently selected. Second, we should bulletproof against exceptions through some combination of making sure we aren't referencing through an undefined property and/or catching exceptions and recovering.

I tried to reproduce in Preview 1 and there is an exception, but maybe not in exactly the same place.

**Comment by JonFerraiolo** [comment]
Brad, assigning to you. I think the problem is with partly with code you have worked with a long ago with WidgetProperties.js and ViewLite.js. 

The basic problem here is that upon switching editors, we don't invoke the logic within WidgetProperties.js to respond to a new widget selection (function onWidgetSelectionChange()). We probably have similar problems with CommonProperties.js and EventSelection.js.

Upon opening a new editor, we don't seem to be firing a /davinci/ui/widgetSelected event. I tried to add the following 3 lines to VisualEditor.js at the end of _editorSelected():

```
        var context = this.getContext();
        var selection = context.getSelection();
        dojo.publish("/davinci/ui/widgetSelected",[selection]);
```

This fired the event, but then there is an IF check in ViewLite.js:39 that stops the process of telling WidgetProperties.js that the selection has changed:

```
        if(this._widget == widget && this._subwidget==widget.subwidget)
            return false;
```

I tried to comment out the above code, but it didn't work, and at that point I thought I should assign to you because you can probably figure out how to fix things in a manner that is consistent with design intent.

**Comment by childsb** [comment]
Fixed in commit 021c68acbddeee4fcb6a8af0e2575fa855075ddd
